examples/streaming/kafka/src/main/scala/org/apache/gearpump/streaming/examples/kafka/KafkaReadWrite.scala

```diff
@@ -86,4 +90,30 @@ object KafkaReadWrite extends AkkaApp with ArgumentsParser {
     val appId = context.submit(application(config, context.system))
     context.close()
   }
+
+
+  class EventTimeKafkaMessageDecoder extends KafkaMessageDecoder {
+    private var count = 0
+    private var localMin = Long.MaxValue
+    private var watermark = 0L
+    private val LOG = LogUtil.getLogger(classOf[EventTimeKafkaMessageDecoder])
+    /**
+     * @param key key of a kafka message, can be NULL
+     * @param value value of a kafka message
+     * @return a gearpump Message
+     */
+    override def fromBytes(key: Array[Byte], value: Array[Byte]): MessageAndWatermark = {
+      Injection.invert[Long, Array[Byte]](key).map { eventTime =>
+        if (count == 10) {
+          watermark = localMin
```

Comment:
shall we consider some delay on watermark? E.g. if Source get messages at 10:00, shall we report that the watermark is 9:55 instead of 10:00 as data is out of order and may be delayed.